h2o-py/h2o/utils/shared_utils.py

```diff
@@ -353,6 +360,129 @@ def slice_is_normalized(s):
+    try:
+        json_object = json.loads(json_str)
+    except ValueError:
+        raise RuntimeError("Error: Given JSON string does not look like valid JSON string.")
+    return True
+
+def predict_json(mojo_model, json, genmodelpath=None, labels=False, classpath=None, javaoptions='-Xmx4g', show_debug=False):
+    """
+    MOJO scoring function to take a pandas data frame as json string and use MOJO model as zip file to score
+
+    :param mojo_model: This is the MOJO model file name which is download after model was build using H2O, you can pass it two ways:
+           1. Full mojo model path i.e. /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring/gbm_prostate_new.zip
+           2. mojo name first i.e. gbm_prostate_new.zip and genmodelpath = /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring
+    :param json:  convert pandas frame to json (pd.to_json) and pass here
+    :param genmodelpath: This is the local file system folder name where both h2o-genmodel.jar and MOJO zip file is stored.
+    :param labels:     (Optional) True : Shows results, False: Does no show results and just pass result to given object
+    :param javaoptions: (Optional) These are the Java string options given by the user
+    :param show_debug: True/False - Default ( Use True to see the full Java command line)
+    :return: score as json values
+    """
+    # Verifying JSON
+    if show_debug:
+        print(json)
+
+    check_json(json)
+
+    # Checking java
+    java = H2OLocalServer._find_java()
+
+    jver_bytes = subprocess.check_output([java, "-version"], stderr=subprocess.STDOUT)
+    jver = jver_bytes.decode(encoding="utf-8", errors="ignore")
+    if show_debug:
+        print("  Java Version: " + jver.strip().replace("\n", "; "))
+    if "GNU libgcj" in jver:
+        raise H2OStartupError("Sorry, GNU Java is not supported for H2O.\n"
+                              "Please download the latest 64-bit Java SE JDK from Oracle.")
+    if "Client VM" in jver:
+        warn("  You have a 32-bit version of Java. H2O works best with 64-bit Java.\n"
+             "  Please download the latest 64-bit Java SE JDK from Oracle.\n")
+
+    # genmodelpath > must start and ends with "/" or add to it
+    gen_model_arg = ".:"
+
+    ## Working on mojo_model
+    mojo_model_path=None
+    head, tail = ntpath.split(mojo_model)
+    if not head:
+        if not genmodelpath:
+            raise RuntimeError("Error: given mojo model " + mojo_model + " does not have full path!")
+        else:
+            mojo_model_path = genmodelpath
+    else:
+        mojo_model_path = head
+
+    if not mojo_model_path:
+        raise RuntimeError("Error: given mojo model " + mojo_model + " does not have full path!")
+
+    # mojo model name is
+    mojo_model_name = tail
+
+
+    # Mojo model is given by name only and separate path is given also
+    # Check both genmodel and pojo is available in the same path
```

Comment:
I think this could be simplified. Behavior of the R counterpart is this:

- if genmodelpath is given it is used
- if genmodelpath is not given it set to the directory where the the mojo is store

This seems straightforward and easy to understand to the user.

Reply:
done. now it use os.path.abspath to get the path from given mojo.